airflow/providers/amazon/aws/hooks/base_aws.py

```diff
@@ -47,11 +48,29 @@ class AwsBaseHook(BaseHook):
     :param verify: Whether or not to verify SSL certificates.
         https://boto3.amazonaws.com/v1/documentation/api/latest/reference/core/session.html
     :type verify: str or bool
+    :param str region_name: AWS Region name to use. If this is None then the default boto3
+        behaviour is used.
+    :param str client_type: boto3 client_type used when creating boto3.client(). For
+        example, 's3', 'emr', etc. Provided by specific hooks for these clients which
+        subclass AwsBaseHook.
+    :param str resource_type: boto3 resource_type used when creating boto3.resource(). For
+        example, 's3'. Provided by specific hooks for these resources which
+        subclass AwsBaseHook.
```

Comment:
Can you use `:param` + `:type` ?

Also the type is `Optional[str]` not `str` because it allows a string to be None. 
You can also define the types directly next to the argument like this: `region_name: Optional[str] = None`. You only need to import `typing`. It is an official PEP https://www.python.org/dev/peps/pep-0484/ in Python >=3.5

Reply:
Resolved, thanks.
I preferred the conciseness of :param <type> <var>: xyz
But have changed back to :param + :type